llama_index/vector_stores/google/generativeai/genai_extension.py

```diff
@@ -461,50 +467,82 @@ def query_document(
 @dataclass
 class Passage:
     text: str
-    ids: List[str]
+    id: str
 
 
 @dataclass
-class TextAnswer:
+class GroundedAnswer:
     answer: str
     attributed_passages: List[Passage]
     answerable_probability: Optional[float]
 
 
-def generate_text_answer(
+@dataclass
+class GenerateAnswerError(Exception):
+    finish_reason: genai.Candidate.FinishReason
```

Comment:
nit: I noticed sometimes we can use the type of the enum and sometimes we use int. Are there some rules around when we can do one vs the other or can we consistently use the enum type?